Okay, we're getting relevant buildbot failures. E.g., from [here](https://buildbot.python.org/all/#/builders/577/builds/1231/steps/4/logs/stdio)

```
======================================================================
FAIL: test_nan_signs (test.test_float.InfNanTest.test_nan_signs)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "D:\buildarea\pull_request.bolen-windows10\build\Lib\test\test_float.py", line 1045, in test_nan_signs
    self.assertEqual(copysign(1.0, float('nan')), 1.0)
AssertionError: -1.0 != 1.0
----------------------------------------------------------------------
```

Looks like we _will_ need that `fabs` call around the `NAN` macro (which isn't really surprising - there's nothing in the C standard that says that `NAN` should have any particular sign bit).

The question is whether we want to put the `fabs` in _just_ for the `float("nan")` and `math.nan` uses (and complex equivalents) and leave `Py_NAN` as the "system" NaN, or include the `fabs` call in the `Py_NAN` macro.

--- a/Include/pymath.h
+++ b/Include/pymath.h
@@ -52,16 +52,11 @@
 #  define Py_HUGE_VAL HUGE_VAL
 #endif
 
-// Py_NAN: Value that evaluates to a quiet and positive Not-a-Number (NaN).
+/* Py_NAN: Value that evaluates to a quiet Not-a-Number (NaN).  The sign is
+ * undefined and normally not relevant, but e.g. fixed for float("nan").
+ */
 #if !defined(Py_NAN)
-#  if _Py__has_builtin(__builtin_nan)
-     // Built-in implementation of the ISO C99 function nan(): quiet NaN.
-#    define Py_NAN (__builtin_nan(""))
-#else
-     // Use C99 NAN constant: quiet Not-A-Number.
-     // NAN is a float, Py_NAN is a double: cast to double.
 #    define Py_NAN ((double)NAN)
-#  endif
 #endif
 
 #endif /* Py_PYMATH_H */

--- a/Modules/cmathmodule.c
+++ b/Modules/cmathmodule.c
@@ -1235,10 +1235,10 @@ cmath_exec(PyObject *mod)
                            PyComplex_FromCComplex(infj)) < 0) {
         return -1;
     }
-    if (PyModule_AddObject(mod, "nan", PyFloat_FromDouble(Py_NAN)) < 0) {
+    if (PyModule_AddObject(mod, "nan", PyFloat_FromDouble(fabs(Py_NAN))) < 0) {
         return -1;
     }
-    Py_complex nanj = {0.0, Py_NAN};
+    Py_complex nanj = {0.0, fabs(Py_NAN)};
     if (PyModule_AddObject(mod, "nanj", PyComplex_FromCComplex(nanj)) < 0) {
         return -1;
     }

--- a/Modules/mathmodule.c
+++ b/Modules/mathmodule.c
@@ -3949,7 +3949,7 @@ math_exec(PyObject *module)
     if (PyModule_AddObject(module, "inf", PyFloat_FromDouble(Py_INFINITY)) < 0) {
         return -1;
     }
-    if (PyModule_AddObject(module, "nan", PyFloat_FromDouble(Py_NAN)) < 0) {
+    if (PyModule_AddObject(module, "nan", PyFloat_FromDouble(fabs(Py_NAN))) < 0) {
         return -1;
     }
     return 0;

--- a/Objects/floatobject.c
+++ b/Objects/floatobject.c
@@ -2430,7 +2430,7 @@ PyFloat_Unpack2(const char *data, int le)
         }
         else {
             /* NaN */
-            return sign ? -Py_NAN : Py_NAN;
+            return sign ? -fabs(Py_NAN) : fabs(Py_NAN);
         }
     }
 

--- a/Python/pystrtod.c
+++ b/Python/pystrtod.c
@@ -46,7 +46,7 @@ _Py_parse_inf_or_nan(const char *p, char **endptr)
     }
     else if (case_insensitive_match(s, "nan")) {
         s += 3;
-        retval = negate ? -Py_NAN : Py_NAN;
+        retval = negate ? -fabs(Py_NAN) : fabs(Py_NAN);
     }
     else {
         s = p;

